ext/standard/streamsfuncs.c

```diff
@@ -1255,8 +1261,8 @@ PHP_FUNCTION(stream_filter_remove)
 
 	filter = zend_fetch_resource(Z_RES_P(zfilter), NULL, php_file_le_stream_filter());
 	if (!filter) {
-		php_error_docref(NULL, E_WARNING, "Invalid resource given, not a stream filter");
-		RETURN_FALSE;
+		zend_value_error("Invalid resource given, not a stream filter");
```

Comment:
TypeError? Or better, pass resource name to zend_fetch_resource and let it throw.